src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,184 @@
+     *
+     * @param isInertiaBased is the configuration for the state space.
+     *                       If the value is true, the state space is based off of an inertia model.
+     *                       Otherwise, the state space is based off of a voltage equation model.
+     * @return the linear system loop (based on the type).
+     */
+    private LinearSystemLoop<N1, N1, N1> configStateSpace(boolean isInertiaBased) {
+        final DCMotor motor = DCMotor.getFalcon500(1);
+
+        LinearSystem<N1, N1, N1> flywheel_plant;
+        if (isInertiaBased) {
+            flywheel_plant = LinearSystemId.createFlywheelSystem(motor, J, GEAR_RATIO);
+        } else {
+            flywheel_plant = LinearSystemId.identifyVelocitySystem(Kv, Ka);
+        }
+        if (Robot.isSimulation()) {
+            flywheelSim = new FlywheelSim(
+                    flywheel_plant,
+                    motor,
+                    GEAR_RATIO);
+        }
+
+        LinearQuadraticRegulator<N1, N1, N1> quadraticRegulator = new LinearQuadraticRegulator<>(
+                flywheel_plant,
+                VecBuilder.fill(VELOCITY_TOLERANCE),
+                VecBuilder.fill(COST_LQR),
+                LOOP_PERIOD);
+        quadraticRegulator.latencyCompensate(flywheel_plant, LOOP_PERIOD, TALON_TIMEOUT / 1000.0);
+
+        KalmanFilter<N1, N1, N1> kalmanFilter = new KalmanFilter<>(
+                Nat.N1(), Nat.N1(),
+                flywheel_plant,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(SENSOR_TOLERANCE),
+                LOOP_PERIOD);
+
+        return new LinearSystemLoop<>(
+                flywheel_plant,
+                quadraticRegulator,
+                kalmanFilter,
+                NOMINAL_VOLTAGE, LOOP_PERIOD);
+    }
+
+    /**
+     * Gets the velocity of the motor.
+     *
+     * @return the velocity of the motor. [rpm]
+     */
+    public double getVelocity() {
+        if (Robot.isSimulation()) {
+            return flywheelSim.getAngularVelocityRPM();
+        }
+        return unitModel.toVelocity(motor.getSelectedSensorVelocity());
+    }
+
+    /**
+     * Sets the velocity of the motor.
+     *
+     * @param velocity is the velocity setpoint. [rpm]
+     */
+    public void setVelocity(double velocity) {
+        linearSystemLoop.setNextR(VecBuilder.fill(Units.rotationsToRadians(Utils.secondsToMinutes(velocity))));
+        linearSystemLoop.correct(VecBuilder.fill(Units.rotationsToRadians(Utils.secondsToMinutes(getVelocity()))));
+        linearSystemLoop.predict(currentTime - lastTime);
+        if (Robot.isSimulation()) {
+            SmartDashboard.putNumber("ve", velocity);
+            flywheelSim.setInputVoltage(MathUtil.clamp(linearSystemLoop.getU(0), -NOMINAL_VOLTAGE, NOMINAL_VOLTAGE));
+            simCollection.setIntegratedSensorVelocity(unitModel.toTicks100ms(Utils.secondsToMinutes(flywheelSim.getAngularVelocityRPM())));
```

Comment:
the conversion doesn't seem to be right (you convert minutes to hours)

Reply:
But I divide by 60

Reply:
WHERE?

Reply:
So? the units should still be in rpm

Reply:
Usually converting minutes to seconds is multiplying by 60. However, because we have sec^-1 and minutes^-1, we need to divide.

Reply:
Should I rename the function to rpmToRps?